packages/effect/test/Effect/promise.test.ts

```diff
@@ -17,7 +17,7 @@ describe("Effect", () => {
     })
     const program = effect.pipe(
       Effect.timeout("10 millis"),
```

Comment:
For all tests / library code that was previously using `Effect.timeout(...).pipe(Effect.optionFromOptional)`, we should be explicit about using `Effect.asSome` for the happy path.

Similar to what was done in the `NodeClient.test.ts` module.